ghost/core/core/server/services/email-analytics/lib/queries.js

```diff
@@ -14,30 +13,36 @@ module.exports = {
     async getLastSeenEventTimestamp() {
         const startDate = new Date();
 
-        // three separate queries is much faster than using max/greatest (with coalesce to handle nulls) across columns
-        let {maxDeliveredAt} = await db.knex('email_recipients').select(db.knex.raw('MAX(delivered_at) as maxDeliveredAt')).first() || {};
-        let {maxOpenedAt} = await db.knex('email_recipients').select(db.knex.raw('MAX(opened_at) as maxOpenedAt')).first() || {};
-        let {maxFailedAt} = await db.knex('email_recipients').select(db.knex.raw('MAX(failed_at) as maxFailedAt')).first() || {};
-
-        if (maxDeliveredAt && !(maxDeliveredAt instanceof Date)) {
-            // SQLite returns a string instead of a Date
-            maxDeliveredAt = new Date(maxDeliveredAt);
-        }
-
-        if (maxOpenedAt && !(maxOpenedAt instanceof Date)) {
-            // SQLite returns a string instead of a Date
-            maxOpenedAt = new Date(maxOpenedAt);
+        // First, try to fetch the max latest_event_timestamp
+        let {maxTimestamp} = await db.knex('emails').select(db.knex.raw('MAX(latest_event_timestamp) as maxTimestamp')).first() || {};
+
+        if (maxTimestamp) {
+            if (!(maxTimestamp instanceof Date)) {
+                maxTimestamp = new Date(maxTimestamp);
+            }
+        } else {
+            let {maxDeliveredAt} = await db.knex('email_recipients').select(db.knex.raw('MAX(delivered_at) as maxDeliveredAt')).first() || {};
+            let {maxOpenedAt} = await db.knex('email_recipients').select(db.knex.raw('MAX(opened_at) as maxOpenedAt')).first() || {};
+            let {maxFailedAt} = await db.knex('email_recipients').select(db.knex.raw('MAX(failed_at) as maxFailedAt')).first() || {};
+
+            if (maxDeliveredAt && !(maxDeliveredAt instanceof Date)) {
+                // SQLite returns a string instead of a Date
+                maxDeliveredAt = new Date(maxDeliveredAt);
+            }
+
+            if (maxOpenedAt && !(maxOpenedAt instanceof Date)) {
+                // SQLite returns a string instead of a Date
+                maxOpenedAt = new Date(maxOpenedAt);
+            }
+
+            if (maxFailedAt && !(maxFailedAt instanceof Date)) {
+                // SQLite returns a string instead of a Date
+                maxFailedAt = new Date(maxFailedAt);
+            }
```

Comment:
I'm guessing we're keeping the old logic to smoothen the rollout, as existing emails won't have the `latest_event_timestamp` field. Shall we make a note to clean this up in a couple of weeks from now?

Reply:
You could also return early here, to separate the new logic from the old one, e.g.

```
if (maxTimestamp) {
    if (!(maxTimestamp instanceof Date)) {
        maxTimestamp = new Date(maxTimestamp);
     }
              
     return maxTimestamp;
}


// Legacy logic

...
```

Reply:
Yes correct, I'll add a note in the comments.